README.md

```diff
@@ -1,66 +1,86 @@
-       URLs that can be downloaded or cloned into the local repository into a
-       duniverse/ subdirectory. Once the external code has been pulled into
-       the repository, a single dune build command is sufficient to build the
-       whole project in a standalone fashion, without opam being required.
-       This is a particularly convenient way of publishing CLI tools to users
-       who do not need the full power of opam.
+This will generate a project-wide lockfile under `<project-name>.opam.locked` that contains:
+- The full list of your direct and transitive opam dependencies, according to the specifications
+  in the `.opam` files at your project's root, with hard version constraints (`{= <version>`) in
+  the `depends` field.
+- A `pin-depends` field filled with reproducible pins, either to tarballs or to git repos with an
+  explicit commit hash for all the above dependencies, except for `ocaml`, `dune` and any virtual
+  opam package.
+- A few extra fields meant to be interprated by `opam monorepo`
 
-       You can access the functionality directly via the monorepo-lock and
-       monorepo-pull commands,
+This lockfile can then be consumed by the following command:
+```
+opam monorepo pull
+```
+which will fetch the sources of all the locked dependencies into a `duniverse/` folder at the root of
+your project, marking them as `vendored_dirs` (see
+[dune's documentation](https://dune.readthedocs.io/en/latest/dune-files.html#vendored-dirs-since-1-11))
+so that dune will only build the artifacts you need from that folder.
 
-       Also see https://github.com/avsm/platform for an example of a fully
-       bootstrapping use of this tool.
+From that point you should be able to run `dune build` and `dune runtest` as you normally would and
+build your entire project from scratch!
 
-COMMANDS
-       lock
-           analyse opam files to generate a project-wide lock file
+### opam monorepo lock
 
-       pull
-           fetch the dependencies sources as specified by the lockfile
+It's important to note that `opam monorepo lock` will only succeed if all of your non-virtual and
+non "base" dependencies (e.g. `ocaml` or `dune`) build with dune (i.e. directly depend on the `dune`
+or `jbuilder` packages).
+If that's not the case the solver will report which packages don't build with dune.
 
-COMMON OPTIONS
-       --help[=FMT] (default=auto)
-           Show this help in format FMT. The value FMT must be one of `auto',
-           `pager', `groff' or `plain'. With `auto', the format is `pager` or
-           `plain' whenever the TERM env var is `dumb' or undefined.
+We maintain a [separate opam repository](https://github.com/dune-universe/opam-overlays) with dune
+ports of commonly used opam packages. If you have non-dune dependencies it is advised that you add
+this repository before running `opam monorepo lock`. You can do so by running the following command:
```

Comment:
I think it's worth mentioning that opam configuration (local pins, remotes) influences the lock file generation, but `pull` does not depend on that.

Reply:
You're right it probably needs clarification!

Reply:
Just pushed a little something!